modules/mux/targets/riscv/source/device_info.cpp

```diff
@@ -70,6 +70,9 @@ device_info_s::device_info_s()
 
   // Override this default from hal updates
   this->address_capabilities = mux_address_capabilities_bits64;
+  this->atomic_capabilities =
+      mux_atomic_capabilities_8bit | mux_atomic_capabilities_16bit |
+      mux_atomic_capabilities_32bit | mux_atomic_capabilities_64bit;
```

Comment:
I think we only support 64 bit atomics if it's a a 64 bit risc-v? (Technically also if we support atomics)

Reply:
This is an existing problem in `device_info_s` that I just left as is: we also set `integer_capabilities` to include `mux_integer_capabilities_64bit` even on 32-bit RISC-V where it doesn't work. It should be aligned with that, I think: on RISC-V, we can support 64-bit atomics in exactly the same cases that we can 64-bit integers in general.

Reply:
I think atomics is a bit different though, we'd generally expect 64 bit atomics on a 64 bit (address space) risc-v. If you tried this code on a 32 bit version, we'll get additional atomic link issues most likely.

Reply:
Per offline discussion, even though 32-bit RISC-V support is currently broken when it comes to 64-bit types, we do want to support that and this PR should be so that it does not make things worse there. Will update.